pkg/pillar/types/ledmanagertypes.go

```diff
@@ -3,17 +3,69 @@
 
 package types
 
+// LedBlinkCount is enum type summarizing all LED blinking patterns.
+type LedBlinkCount uint8
+
+const (
+	// Undefined LED blinking pattern.
+	LedBlinkUndefined LedBlinkCount = iota
+	// LED indication of device waiting to obtain management IP address.
+	LedBlinkWaitingForIP
+	// LED indication of device trying to connect to the controller.
+	LedBlinkConnectingToController
+	// LED indication of device being connected to the controller but not yet onboarded.
+	LedBlinkConnectedToController
+	// LED indication of device being connected to the controller and onboarded.
+	LedBlinkOnboarded
+)
+const (
+	// LED indication of device failing to onboard.
+	LedBlinkOnboardingFailure LedBlinkCount = iota + 10
+	_                                       // 11 is unused
+	// LED indication or device receiving response from controller without TLS connection state.
+	LedBlinkRespWithoutTLS
+	// LED indication or device receiving response from controller without OSCP.
+	LedBlinkRespWithoutOSCP
+	// LED indication or device failing to validate or fetch the controller certificate.
+	LedBlinkInvalidControllerCert
+)
+
+func (c LedBlinkCount) String() string {
+	switch c {
+	case LedBlinkUndefined:
+		return "undefined state"
+	case LedBlinkWaitingForIP:
+		return "waiting for IP address"
+	case LedBlinkConnectingToController:
+		return "connecting to controller"
+	case LedBlinkConnectedToController:
+		return "connected to controller"
+	case LedBlinkOnboarded:
+		return "onboarded"
+	case LedBlinkOnboardingFailure:
+		return "onboarding failure"
+	case LedBlinkRespWithoutTLS:
+		return "response without TLS"
+	case LedBlinkRespWithoutOSCP:
+		return "response without OSCP"
+	case LedBlinkInvalidControllerCert:
+		return "invalid controller certificate"
+	default:
+		return "unrecognized state"
```

Comment:
fmt.Sprintf with %d makes this easier to find.

Reply:
done